### PR TITLE
Rename LAR to LARS

### PR DESCRIPTION
**Pull request by jschueller**
I propose we leave it deprecated during 2 versions instead of one as it's widely used.

### CODE REVIEW

**Comment by regislebrun** [comment]
+1

File: lib/src/Base/Func/openturns/LARS.hxx [hunk 1]
```diff
@@ -11,15 +11,15 @@
  *
  *  This library is distributed in the hope that it will be useful,
  *  but WITHOUT ANY WARRANTY; without even the implied warranty of
- *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULARS PURPOSE.  See the
  *  GNU Lesser General Public License for more details.
```
Comment (sofianehaddad) [inline lib/src/Base/Func/openturns/LARS.hxx]: care you changed the terms of license

File: lib/src/Base/Func/LARS.cxx [hunk 2]
```diff
@@ -11,7 +11,7 @@
  *
  *  This library is distributed in the hope that it will be useful,
  *  but WITHOUT ANY WARRANTY; without even the implied warranty of
- *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULARS PURPOSE.  See the
  *  GNU Lesser General Public License for more details.
```
Comment (sofianehaddad) [inline lib/src/Base/Func/LARS.cxx]: same here

File: ChangeLog [hunk 3]
```diff
@@ -33,6 +33,7 @@
  * Deprecated classes UserDefinedPair, HistogramPair
  * Removed SensitivityAnalysis class
  * Deprecated SLSQP, LBFGS and NelderMead classes in favor of NLopt class
+ * Deprecated LAR in favor of LARS
```
Comment (sofianehaddad) [inline ChangeLog]: you don't exactly deprecate the LAR class, you rename it into LARS.
Question : how to still use LAR until 1.10?

Comment (jschueller) [inline ChangeLog]: @sofianehaddad, it's renamed in the c++, but I added LAR(LARS) in lars.i, I should have mentioned that

Comment (sofianehaddad) [inline ChangeLog]: @jschueller ok sorry did not see it.

**Comment by sofianehaddad** [comment]
+1

**Pull request merged**